ui/src/lib/features/k8s/store.ts

```diff
@@ -167,13 +167,20 @@ export class ResourceStore<T extends KubernetesObject, U extends CommonRow> impl
    * @returns A function to stop the EventSource
    */
   start() {
-    // If the store has already been initialized, return
     if (this.#initialized) {
       return () => {}
     }
 
     this.#initialized = true
-    this.#eventSource = new EventSource(this.url)
+
+    if (!get(apiAuthEnabled)) {
+      this.#eventSource = new EventSource(this.url)
+    } else {
+      const apiToken: string = sessionStorage.getItem('token') ?? ''
+      // Check if the URL already contains a '?' for urls with multiple search params
+      const separator = this.url.includes('?') ? '&' : '?'
+      this.#eventSource = new EventSource(`${this.url}${separator}token=${apiToken}`)
+    }
```

Comment:
what do you think about something like this to get rid of the negative assertion and simplify a bit?

```suggestion
   if (get(apiAuthEnabled)) {
      const apiToken: string = sessionStorage.getItem('token') ?? ''
      // Check if the URL already contains a '?' for urls with multiple search params
      const separator = this.url.includes('?') ? '&' : '?'
      this.url = `${this.url}${separator}token=${apiToken}`
    }
    
    this.#eventSource = new EventSource(this.url)
```